static/src/javascripts/projects/common/modules/commercial/braze/getMessageFromUrlFragment.ts

```diff
@@ -0,0 +1,65 @@
+import { log } from '@guardian/libs';
+import type { BrazeMessageInterface } from './brazeMessageInterface';
+
+const FORCE_BRAZE_ALLOWLIST = [
+	'preview.gutools.co.uk',
+	'preview.code.dev-gutools.co.uk',
+	'localhost',
+	'm.thegulocal.com',
+];
+
+const isExtrasData = (data: unknown): data is Record<string, string> => {
+	if (typeof data === 'object' && data != null) {
+		return Object.values(data).every((value) => typeof value === 'string');
+	}
+	return false;
+};
```

Comment:
@guardian/libs has an `isObject` and `isString` method, which you may want to use for consistency.